ludwig/utils/fs_utils.py

```diff
@@ -51,9 +51,23 @@ def path_exists(url):
     return fs.exists(path)
 
 
+def rename(src, tgt):
+    protocol, _ = split_protocol(tgt)
+    if protocol is not None:
+        fs = fsspec.filesystem(protocol)
+        fs.mv(src, tgt, recursive=True)
+    else:
+        os.rename(src, tgt)
+
+
 def makedirs(url, exist_ok=False):
     fs, path = get_fs_and_path(url)
-    return fs.makedirs(path, exist_ok=exist_ok)
+    if fs == "s3":
```

Comment:
Not sure why this is needed for s3, since directories don't exist. Is this so checks like `exists(dirname)` will pass?

This may also be true for other object stores. Maybe something more generic we could do here is:

```
fs.makedirs(path, exist_ok=exist_ok)
if not exists(paths):
    # Directory was not created -> no directory concept in filesystem
    with fsspec.open(url, mode="wb") as f:
            pass
```

What do you think?